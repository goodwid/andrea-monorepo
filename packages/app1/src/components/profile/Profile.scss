@import 'shared/styles/variables.scss';

:local(.Profile) {
  position: absolute;
  top: $header-size;
  width: 100%;
  height: $content-size;

  font-family: $default-font;
  overflow: scroll;

  .profile-header {
    display: grid;
    grid-template-rows: 1fr 1fr;
    text-align: center;
    align-items: center;
    justify-content: center;
    background-color: #f3f3f3;
    width: 100vw;
    padding: 0.4rem;
    min-height: 9rem;

    .edit {
      position: absolute;
      top: 1.4rem;
      right: 0.8rem;
      font-size: 1.6rem;
    }

    .settings {
      position: absolute;
      top: 1.4rem;
      left: 0.8rem;
      font-size: 1.6rem;
    }

    .name {
      margin: 0.4rem;
      font-size: 1.9rem;
      font-weight: 700;
    }

    .description {
      font-size: 1.6rem;
    }
  }

  .details-section {
    width: 100vw;

    span {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 5rem;
      width: 100%;
      font-weight: 500;
      font-size: 1.8rem;
      border-bottom: 3px $light-gray solid;
    }

    .bio-section {
      display: grid;
      grid-gap: 1rem;
      text-align: center;
      align-items: center;
      width: 90vw;
      margin: 1rem auto;

      p {
        margin: 1rem;
        font-size: 1.4rem;
        text-align: center;
        line-height: 1.4;
      }

      h3 {
        font-weight: 500;
        font-size: 1.6rem;
      }

      .account-holder-name {
        margin: 0rem;
      }
    }
  }
}
