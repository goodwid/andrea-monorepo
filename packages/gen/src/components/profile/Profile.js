import React from 'react';
import styles from './Profile.scss';
import Stars from 'shared/components/stars/Stars';

// 3RD PARTY LIBRARIES
import {
  FaEdit as Edit,
  FaCog as Settings,
} from 'react-icons/fa';

const Profile = () => {
  return (
    <article className={styles.Profile}>
      <section className="profile-header">
        <Edit className="edit" />
        <Settings className="settings" />
        <h1 className="name">Your Name</h1>
        <h2 className="description">Description</h2>
      </section>
      <section className="details-section">
        <Stars rating={3}/>
        <section className="bio-section">
          <h3>Bio</h3>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
        </section>
      </section>
    </article>
  );
};

export default Profile;
