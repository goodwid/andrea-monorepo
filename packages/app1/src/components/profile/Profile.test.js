import React from 'react';
import { shallow } from 'enzyme';
import Profile from './Profile';

describe('Profile page', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<Profile/>);
    expect(wrapper).toMatchSnapshot();
  });
});
