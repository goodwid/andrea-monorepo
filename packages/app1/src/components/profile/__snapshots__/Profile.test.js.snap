// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page matches the snapshot 1`] = `
<article
  className="Profile"
>
  <section
    className="profile-header"
  >
    <FaEdit
      className="edit"
    />
    <FaCog
      className="settings"
    />
    <h1
      className="name"
    >
      Your Name
    </h1>
    <h2
      className="description"
    >
      Description
    </h2>
  </section>
  <section
    className="details-section"
  >
    <Stars
      rating={3}
    />
    <section
      className="bio-section"
    >
      <h3>
        Bio
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
    </section>
  </section>
</article>
`;
